numpy/random/mtrand.pyx

```diff
@@ -724,17 +723,7 @@ cdef class RandomState:
             high = low
             low = 0
 
-        dt = np.dtype(dtype)
-        key = dt.name
-        if key not in _integers_types:
-            raise TypeError('Unsupported dtype "%s" for randint' % key)
-        if not dt.isnative:
-            # numpy 1.17.0, 2019-05-28
-            warnings.warn('Providing a dtype with a non-native byteorder is '
```

Comment:
I think you can keep this and add `dt = dt.newbyteorder()`

Reply:
Ok, good idea.